### PR TITLE
💻 Add Reviews to Resources

### LINKED ISSUE #163: Display reviews
Review consists of:
- comment
- rating (each resource will have different axis to rate on)

Requirements:
- display reviews

## Rating Axis'
Housing:
- cleanliness
- price
- company

Study Spots:
- cleanliness
- noise

Washrooms:
- cleanliness

### PR DESCRIPTION
**Pull request by a22yuen**
## Screenshots
On laptops/desktops:
image

On mobile: 
image

## Purpose
Adds reviews to each resource. Closes #163 

## Approach
Changed types of Resources. This includes adding different parameters for the ratings based on the type of resource. Also add cards to display each review

Used [react-stars](https://www.npmjs.com/package/react-stars) for rating UI.

### CODE REVIEW

File: client/components/resource-details/ResourceInfo.tsx [hunk 1]
```diff
@@ -144,29 +137,164 @@ const Reviews = ({ reviews }: ReviewsProps) => {
...
+  margin-right: 5px;
+  padding-left: auto;
+  flex: 1;
+  flex-grow: 1;
+  font-weight: 600;
+  width: auto;
+  color: ${colours.neutralDark2};
+`
+
+const Rating = styled(({ className, rating }: { className?: string; rating: [string, number] }) => {
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:275]: Can we pass `rating` and `number` in as separate params here? Using array indexing to access them feels weird

File: client/components/resource-details/ResourceInfo.tsx [hunk 2]
```diff
@@ -144,29 +137,164 @@ const Reviews = ({ reviews }: ReviewsProps) => {
...
+  }
+  interval = seconds / 3600
+  if (interval > 2) {
+    return Math.floor(interval) + ' hours'
+  }
+  interval = seconds / 60
+  if (interval > 2) {
+    return Math.floor(interval) + ' minutes'
+  }
+  return Math.floor(seconds) + 2 + ' seconds'
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:230]: There's a library for that
https://date-fns.org/v2.22.1/docs/formatRelative

File: client/components/resource-details/ResourceInfo.tsx [hunk 3]
```diff
@@ -144,29 +137,164 @@ const Reviews = ({ reviews }: ReviewsProps) => {
...
+`
+
+const ShowOnLaptop = styled.div`
+  @media ${smallerThan(width.laptop)} {
+    display: none;
+  }
 `
 
-const Rating = styled.p``
+const HideOnLaptop = styled.div`
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:182]: I really like this idea, we might even want to factor this into a util file under `styles` someday

File: client/components/resource-details/ResourceInfo.tsx [hunk 4]
```diff
@@ -95,46 +102,32 @@ export const ResourceInfo = ({ resource }: ResourceInfoProps) => {
...
   const images = links.map((link) => {
     return {
       original: link,
       thumbnail: 'https://picsum.photos/id/1018/250/150/',
     }
   })
-
   return (
-    <GalleryWrapper>
+    <div className={className}>
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:118]: IIRC this is a common React idiom for passing in styles, but we are using `styled-components` so I'm not sure we actually need it

File: client/components/resource-details/ResourceInfo.tsx [hunk 5]
```diff
@@ -144,29 +137,164 @@ const Reviews = ({ reviews }: ReviewsProps) => {
   return (
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:137]: one thing we might support later in this component is sorting/filtering reviews

**Review by xujustinj — COMMENTED** [review]
great responsive design!

File: client/components/resource-details/ResourceInfo.tsx (hunk 5 above)
Comment (evtyz) [reply client/components/resource-details/ResourceInfo.tsx:164]: Added default sorting by timestamp. In the future, we can pass a `sortBy` prop to the `Reviews` component and sort `reviews` according to the prop.

File: client/utils/types.ts [hunk 6]
```diff
@@ -14,22 +14,60 @@ export interface Resource {
...
+  Health,
+  Science,
+  Environment
+}
+
 export interface Review {
   avatarImage: string
   comment: string
-  rating: number
+  timestamp: Date
```
[the remark this answers is not in the document]
Comment (evtyz) [reply client/utils/types.ts:63]: Added date-fns

File: client/components/resource-details/ResourceInfo.tsx (hunk 1 above)
Comment (evtyz) [reply client/components/resource-details/ResourceInfo.tsx]: Done

File: client/components/resource-details/ResourceInfo.tsx (hunk 2 above)
Comment (evtyz) [reply client/components/resource-details/ResourceInfo.tsx]: Replaced this function with date-fns formatRelative, resolved

File: client/components/resource-details/ResourceInfo.tsx [hunk 7]
```diff
@@ -26,15 +28,15 @@ const Name = styled(PageTitle)`
   margin-top: 32px;
 `
```
Comment (jeffzh4ng) [inline client/components/resource-details/ResourceInfo.tsx:30]: Why are we commenting out this code? Can just delete it?

File: client/components/resource-details/ResourceInfo.tsx [hunk 8]
```diff
@@ -1,10 +1,12 @@
...
 import styled from 'styled-components'
-import { colours, fontWeight, PageTitle, width, largerThan } from '../../styles'
+import { colours, fontWeight, PageTitle, largerThan, smallerThan, width } from '../../styles'
 import { Resource } from '../../utils'
-import { Review } from '../../utils/types'
+import { Review, Faculty, Status } from '../../utils/types'
 import { TagBubble } from '../common/TagBubble'
+import ReactStars from 'react-stars'
 import ImageGallery from 'react-image-gallery'
+import { formatRelative } from 'date-fns'
```
Comment (jeffzh4ng) [inline client/components/resource-details/ResourceInfo.tsx:9]: What does this function do? Do we need an entire package just for this? Can we re-implement it ourselves?

**Review by jeffzh4ng — APPROVED** [review]
Overall great job! We should think about splitting up this file into separate components now too.

File: client/components/resource-details/ResourceInfo.tsx [hunk 9]
```diff
@@ -2,7 +2,7 @@ import { useRouter } from 'next/router'
 import styled from 'styled-components'
 import { colours, fontWeight, PageTitle, largerThan, smallerThan, width } from '../../styles'
 import { Resource } from '../../utils'
-import { Review, Faculty } from '../../utils/types'
+import { Review, Faculty, Status } from '../../utils/types'
```
Comment (xujustinj) [inline client/components/resource-details/ResourceInfo.tsx:5]: unused imports, I think

```suggestion
import { Review } from '../../utils/types'
```

File: client/components/resource-details/ResourceInfo.tsx (hunk 8 above)
Comment (xujustinj) [reply client/components/resource-details/ResourceInfo.tsx:9]: The library is a date util library like Moment. See the deleted `timeSince` function below for why we probably shouldn't implement it ourselves -- lots of edge cases. We will have much better correctness guarantees from using a maintained third-party library that undergoes its own testing.

**Review by xujustinj — APPROVED** [review]
very nice

**Review by xujustinj — APPROVED** [review]

**Pull request merged**